pkg/kubectl/cmd/util/factory_object_mapping.go

```diff
@@ -288,19 +287,11 @@ func (f *ring1Factory) Scaler(mapping *meta.RESTMapping) (kubectl.Scaler, error)
 		return nil, err
 	}
 
-	// create scales getter
-	// TODO(p0lyn0mial): put scalesGetter to a factory
-	discoClient, err := f.clientAccessFactory.DiscoveryClient()
-	if err != nil {
-		return nil, err
-	}
-	restClient, err := f.clientAccessFactory.RESTClient()
+	mapper, _ := f.Object()
```

Comment:
The `Scaler` function can temporarily move to `ring2Factory`.  It should have very few callers.  Open an issue to remove all of them.